packages/backend/src/helpers/delay-as-milliseconds.ts

```diff
@@ -1,23 +1,28 @@
-import Step from '@/models/step'
+import { IJSONObject } from '@plumber/types'
 
 import delayForAsMilliseconds, {
   TDelayForUnit,
 } from './delay-for-as-milliseconds'
 import delayUntilAsMilliseconds from './delay-until-as-milliseconds'
 
-const delayAsMilliseconds = (step: Step) => {
+const delayAsMilliseconds = (
+  stepKey: string,
+  computedParameters: IJSONObject,
+) => {
   let delayDuration = 0
 
-  if (step.key === 'delayFor') {
-    const { delayForUnit, delayForValue } = step.parameters
-
+  if (stepKey === 'delayFor') {
+    const { delayForUnit, delayForValue } = computedParameters
     delayDuration = delayForAsMilliseconds(
       delayForUnit as TDelayForUnit,
       Number(delayForValue),
     )
-  } else if (step.key === 'delayUntil') {
-    const { delayUntil } = step.parameters
-    delayDuration = delayUntilAsMilliseconds(delayUntil as string)
+  } else if (stepKey === 'delayUntil') {
+    const { delayUntil, delayUntilTime } = computedParameters
+    delayDuration = delayUntilAsMilliseconds(
+      delayUntil as string,
+      delayUntilTime as string,
+    )
   }
 
   return delayDuration
```

Comment:
Out of scope for this PR but im thinking we should not enqueue the job if delay is negative.

Reply:
hmm yea i saw this occurring but thought that it wasn't much of an issue because it shouldnt break any use case